src/main/java/com/mindee/parsing/common/Execution.java

```diff
@@ -0,0 +1,103 @@
+package com.mindee.parsing.common;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.mindee.product.generated.GeneratedV1Document;
+import java.time.LocalDateTime;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+/**
+ * Representation of a workflow execution.
+ */
+@Getter
+@EqualsAndHashCode
+@JsonIgnoreProperties(ignoreUnknown = true)
+@AllArgsConstructor
+@NoArgsConstructor
+public class Execution<DocT extends Inference> implements ApiObject {
+  /**
+   * Identifier for the batch to which the execution belongs.
+   */
+  @JsonProperty("batch_name")
+  private String batchName;
+
+  /**
+   * The time at which the execution started.
+   */
+  @JsonProperty("created_at")
+  @JsonDeserialize(using = LocalDateTimeDeserializer.class)
+  private LocalDateTime createdAt;
+
+  /**
+   * File representation within a workflow execution.
+   */
+  @JsonProperty("file")
+  private ExecutionFile file;
+
+  /**
+   * Identifier for the execution.
+   */
+  @JsonProperty("id")
+  private String id;
+
+  /**
+   * Deserialized inference object.
+   */
+  @JsonProperty("inference")
+  private DocT inference;
+
+  /**
+   * Priority of the execution.
+   */
+  @JsonProperty("priority")
+  private String priority;
```

Comment:
represent as an enum?

Reply:
I'd rather wait for @thentgesMindee's opinion on the matter

Reply:
probably easier as an enum since on backend side we'll allow only the specific enum values
-> would be better in term of UX and autocompletions to not be able to send anything else with the SDK ?

Reply:
No, I was just worrying about the current enum implementation evolving from what it is now. I'll implement it as an enum then, thanks :+1: